.github/workflows/publish_to_maven_central.yml

```diff
@@ -0,0 +1,45 @@
+name: Publish to Maven Central
+
+on:
+  workflow_call
+
+jobs:
+  publish_to_maven_central:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up JDK
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'zulu'
+          java-version: 17
+
+      - name: Gradle cache
```

Comment:
I know we had it before, but should we remove caching for releases?

Reply:
I think the cache can help us execute gradle tasks faster. I did a small experiment with it.

I just ran different workflows with and without cache.
[Workflow with cache](https://github.com/Adyen/adyen-android/actions/runs/11591406960): It looks like it saves us 3.5 minutes on average.
[Workflow without cache](https://github.com/Adyen/adyen-android/actions/runs/11590456730): The cache has been removed from `check` and `publish_to_maven_central` jobs. The first thing I see is that it downloads the gradle, which IMO takes the extra time. 
![image](https://github.com/user-attachments/assets/f83ead7e-e845-4c68-9361-0d0b25cb4b05)

Reply:
Yes I understand it takes more time, but if the cache is ever corrupt we might publish a build that is broken

Reply:
Hmm, I see your point. I will remove it then so we are safe.